client-misk-jooq/src/main/kotlin/app/cash/backfila/client/misk/jooq/internal/OpenKeyRange.kt

```diff
@@ -0,0 +1,152 @@
+package app.cash.backfila.client.misk.jooq.internal
+
+import app.cash.backfila.client.misk.jooq.JooqBackfill
+import app.cash.backfila.protos.clientservice.GetNextBatchRangeRequest
+import com.squareup.backfila.client.base.jooq.CompoundKeyComparer
+import com.squareup.backfila.client.base.jooq.CompoundKeyComparisonOperator
+import org.jooq.Condition
+import org.jooq.DSLContext
+import org.jooq.Record
+import org.jooq.impl.DSL.select
+
+
+data class OpenKeyRange<K>(
+  private val jooqBackfill: JooqBackfill<K>,
```

Comment:
This coupling seems a bit tight. That said, its probably pretty refactorable.

Reply:
True. I'd like to leave it like this for now. Maybe come back to it if I have sometime left.